### PR TITLE
Moved the About dialog to an activity

### LINKED ISSUE #4225: [Android] Convert 'About' dialog to page for better usability

### PR DESCRIPTION
**Pull request by delta1513**
Fixes #4225

**Description of the Change**
The dialog for the About page has been changed into an activity.

**Release Notes**
N/A

### CODE REVIEW

File: android/BOINC/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,21 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?><!--
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/AndroidManifest.xml:1]: Please keep the copyright header

File: android/BOINC/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -157,4 +155,4 @@
                 android:theme="@style/Theme.AppCompat.DayNight.Dialog" />
     </application>
 
-</manifest>
+</manifest>
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/AndroidManifest.xml:158]: Please add back newline at the end of file

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/AboutActivity.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import edu.berkeley.boinc.R
+import edu.berkeley.boinc.utils.Logging
+
+class AboutActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_about)
+        // about_title is used in AndroidManifest.xml but the compiler doesn't detect it
+        // So we use it here to keep it happy
+        val _about_title = R.string.about_title
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/AboutActivity.kt:19]: This doesn't look correct to me

File: android/BOINC/app/src/main/res/layout/activity_about.xml [hunk 4]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/res/layout/activity_about.xml:1]: Please add copyright header

**Review by AenBleidd — CHANGES_REQUESTED** [review]
Please also fix unit tests

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/AboutActivity.kt (hunk 3 above)
Comment (delta1513) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/AboutActivity.kt]: New commit removes this in order to see if gradle complains about it again or if I was just too tired the first time around

Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/AboutActivity.kt]: If Gradle complains - that could be an issue in Gradle script as well ;)

File: android/BOINC/app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -2,17 +2,14 @@
   This file is part of BOINC.
   http://boinc.berkeley.edu
   Copyright (C) 2021 University of California
-
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/AndroidManifest.xml:5]: Please keep this empty line in between

File: android/BOINC/app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -2,17 +2,14 @@
...
   BOINC is free software; you can redistribute it and/or modify it
   under the terms of the GNU Lesser General Public License
   as published by the Free Software Foundation,
   either version 3 of the License, or (at your option) any later version.
-
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/AndroidManifest.xml:10]: Please keep this empty line in between

File: android/BOINC/app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -2,17 +2,14 @@
...
   BOINC is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
   See the GNU Lesser General Public License for more details.
-
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/AndroidManifest.xml:15]: Please keep this empty line in between

File: android/BOINC/app/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -2,17 +2,14 @@
   This file is part of BOINC.
   http://boinc.berkeley.edu
   Copyright (C) 2021 University of California
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/AndroidManifest.xml:4]: Please change to 2022

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/AboutActivity.kt [hunk 9]
```diff
@@ -0,0 +1,50 @@
+/*
+ * This file is part of BOINC.
+ * http://boinc.berkeley.edu
+ * Copyright (C) 2021 University of California
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/AboutActivity.kt:4]: Please change to 2022

File: android/BOINC/app/src/main/res/layout/activity_about.xml [hunk 10]
```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  This file is part of BOINC.
+  http://boinc.berkeley.edu
+  Copyright (C) 2021 University of California
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/res/layout/activity_about.xml:4]: Please change to 2022

**Review by AenBleidd — CHANGES_REQUESTED** [review]

**Review by AenBleidd — APPROVED** [review]

**Pull request merged**